core/services/beacon/settings.py

```diff
@@ -0,0 +1,110 @@
+import json
+import re
+import socket
+from typing import Any, Dict
+
+import psutil
+from commonwealth.settings import settings
+from pykson import (
+    IntegerField,
+    JsonObject,
+    ListField,
+    MultipleChoiceStringField,
+    ObjectListField,
+    StringField,
+)
+
+
+class DefaultSettings(JsonObject):
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+
+class Interface(JsonObject):
+    name = StringField()
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+    def get_phys(self) -> psutil._common.snicaddr:
+        """
+        return Physical interface from psutil
+        """
+        return psutil.net_if_addrs()[self.name]
+
+    def get_ip_str(self) -> str:
+        """
+        get ip as a string. this also translates 'ips[n]' to the n-th ip in that interface
+        """
+        try:
+            # validate using socket
+            socket.inet_aton(self.ip)
+            return str(self.ip)
+        except OSError:
+            pass
+
+        if "ips[" in self.ip:
+            index = int(re.findall(r"ips\[(\d+)]", self.ip)[0])
+            if not index:
+                index = 0
+            return str(self.get_phys()[index].address)
+        return str(self.get_phys()[0].address)
```

Comment:
```suggestion
        index = 0
        if "ips[" in self.ip:
            matches = re.findall(r"ips\[(\d+)]", self.ip)
            if len(matches) != 0:
                index = int(match)
        return str(self.get_phys()[index].address)
```
This is a safest alternative.

Reply:
yep, much better.

Reply:
you can also remove the first if, since the regex will catch it